packages/aws-cdk-lib/aws-cloudfront-origins/lib/s3-bucket-origin.ts

```diff
@@ -0,0 +1,197 @@
+import { Construct } from 'constructs';
+import * as cloudfront from '../../aws-cloudfront';
+import { AccessLevel } from '../../aws-cloudfront';
+import * as iam from '../../aws-iam';
+import { IBucket } from '../../aws-s3';
+import { Annotations, Aws, Names, Stack } from '../../core';
+
+const BUCKET_ACTIONS = {
+  READ: ['s3:GetObject'],
+  WRITE: ['s3:PutObject'],
+  DELETE: ['s3:DeleteObject'],
+};
+
+/**
+ * Properties for configuring a origin using a standard S3 bucket
+ */
+export interface S3BucketOriginBaseProps extends cloudfront.OriginProps {}
+
+/**
+ * Properties for configuring a S3 origin with OAC
+ */
+export interface S3BucketOriginWithOACProps extends S3BucketOriginBaseProps {
+  /**
+  * An optional Origin Access Control
+  * @default - an Origin Access Control will be created.
+  */
+  readonly originAccessControl?: cloudfront.IOriginAccessControl;
```

Comment:
I found that this interface follows this "extends path":
`S3BucketOriginWithOACProps -> S3BucketOriginBaseProps -> cloudfront.OriginProps -> OriginOptions`

And in `OriginOptions`, there is a prop `originAccessControlId`.  So the user can do:
```
origins.S3BucketOrigin.withOriginAccessControl(bucket, {
          originAccessControlId: 'oacId123',
          originAccessControl: new cloudfront.S3OriginAccessControl(stack, 'MyOac'),
        })
```
Then in the CFN template, the `originAccessControlId` will be ignored.

Reply:
Good catch, we should remove `originAccessControlId` from `OriginOptions` and only allow user to configure OAC with object of type `IOriginAccessControl`. This follows the design guidelines to use strong typing when referencing resources: https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#construct-interface.